Memory_Excercises/Excercise5/Source.cpp

```diff
@@ -0,0 +1,126 @@
+/*Exercise 5 - Dynamic Allocation for Array Sizes
+*Create a Student class that contains a char array of size 5 for a name and an int for an age. Ask
+*the user to enter how many students they want in a class. Then, initialize an array of students in
+*the heap of exactly the size they asked for. You can now probably tell how dynamic allocation is
+*useful.
+*Ask the user for a name and an age for each student and then print them all to the screen. Note
+*that you won�t be able to use names longer than 4 characters.
+*Remember to free any allocated memory.*/
+
+/**/
+
+#include <iostream>
+
+
+class FStudent
+{
+public:
+	char* NameStudent;
+	int StudentAge;
+
+	FStudent()
+	{
+		NameStudent = nullptr;
+		StudentAge = 0;
+	}
+
+	//----------------------------------------------
+	FStudent( int NewAge)
+	{
+		NameStudent = nullptr;
+		StudentAge = NewAge;
+	}
+
+	//----------------------------------------------
+	~FStudent()
+	{
+		delete NameStudent;
+	}
+
+	//----------------------------------------------
+	void SetName(const char* NewName)
+	{
+		/*Exercise 9: At this point, it�s obvious that keeping track of all memory allocations in a program can be a hassle. In the future, 
+		*we�ll discuss techniques that will make life a lot easier for us.
+		*/
+
+		if (NameStudent != nullptr)
+		{
+			NameStudent = nullptr;
```

Comment:
MMmmm, you're actually not fixing the memory leak this way, you should also delete what's saved into this variable before setting it to nullptr